crates/apollo-compiler/src/diagnostic.rs

```diff
@@ -300,24 +300,25 @@ impl ariadne::Cache<FileId> for Cache<'_> {
         }
     }
 
-    fn display<'a>(&self, file_id: &'a FileId) -> Option<Box<dyn fmt::Display + 'a>> {
-        if *file_id != FileId::NONE {
-            struct Path(Arc<SourceFile>);
-            impl fmt::Display for Path {
-                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                    self.0.path().display().fmt(f)
+    fn display<'a>(&self, file_id: &'a FileId) -> Option<impl fmt::Display + 'a> {
+        enum Path {
+            SourceFile(Arc<SourceFile>),
+            NoSourceFile,
+        }
+        impl fmt::Display for Path {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                match self {
+                    Path::SourceFile(source_file) => source_file.path().display().fmt(f),
```

Comment:
this is adding an extra white space around path for some reason 🤔 
so instead of this:
```
Error: `typeFragment1` contains too much nesting
    ╭─[overflow.graphql:11:11]
```

we get this:
```
Error: `typeFragment1` contains too much nesting
    ╭─[ overflow.graphql:11:11 ]
```

Reply:
I think this is another change in Ariadne. We changed snapshots in router: https://github.com/apollographql/router/pull/7012